presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -105,9 +105,13 @@
     private final Deque<Iterator<InternalHiveSplit>> fileIterators = new ConcurrentLinkedDeque<>();
 
     // Purpose of this lock:
+    // * For operations that require exclusive access to either: partitions, fileIterators or hiveSplitSource
+    //   use a write lock
+    // * For checking state use a read lock
```

Comment:
* Write lock: when you need a consistent view across partitions, fileIterators, and hiveSplitSource.
* Read lock: when you need to modify any of the above. Make sure the lock is held throughout the period during which they may not be consistent with each other.

Reply:
Updated.